core/src/test/java/org/apache/calcite/test/SqlValidatorBestEffortTest.java

```diff
@@ -143,4 +143,51 @@
     String expected = "Column 'FOO' not found in any table";
     sql(sql).fails(expected);
   }
+
+  @Test public void testAmbiguousUnknownColumnNotExpanded() {
+    String sql = "SELECT a "
+        + "FROM foo "
+        + "INNER JOIN bar ON foo.x = bar.x "
+        + "INNER JOIN emp on emp.empno = foo.x";
+    String expected = "SELECT `A`\n"
+        + "FROM `FOO` AS `FOO`\n"
+        + "INNER JOIN `BAR` AS `BAR` ON `FOO`.`X` = `BAR`.`X`\n"
+        + "INNER JOIN `CATALOG`.`SALES`.`EMP` AS `EMP`"
+        + " ON `EMP`.`EMPNO` = `FOO`.`X`";
+    sql(sql)
+        .withValidatorIdentifierExpansion(true)
+        .rewritesTo(expected);
+  }
+
+  @Test public void testColumnFromKnownTableExpanded() {
+    String sql = "SELECT empno "
+        + "FROM foo "
+        + "INNER JOIN emp on emp.empno = foo.x";
```

Comment:
Out of curiosity, where are "emp" and "dept" defined as known tables?

Reply:
[MockCatalogReaderSimple](https://github.com/dstekol/calcite/blob/paren_from_clause/core/src/test/java/org/apache/calcite/test/catalog/MockCatalogReaderSimple.java)
Also linked in design doc in test section.

Reply:
Great thanks